### DISCUSSION TITLE
What's the difference of Turbo with NX?

### DISCUSSION THREAD

**Discussion by denik1981** [vercel/turborepo#440] [Help] [answered]
I'm considering switching to Turbo after a of huge pains in trying to set up my NX mono. 

All in all, if your mono follows the use cases of what NX is aimed to solved you are good to go. But if for some reason, you want to move out an inch and improve your mono experience the pain starts to grow. Thus I'm considering moving my entire project to Turbo but need a review from you devs to get a sense of how different it is from NX. There are a lot of similarities in there which makes me think Jared was inspired on it as well but the topology and the end result seems to be much cleaner which I like (based on the examples). Another thing I think is worth credit, is the INCREMENTAL ADOPTION you have built on it. I haven't found the same experience with NX. You need to be a master to start feeling a little bit comfortable with it.

**Comment by alexn-s** [comment, marked as answer]
When i switched to a monorepo solution a few months back i took a look at lerna and nx and a few others (rush, bazel,...) but decied lerna and nx where best suited for me.

At first nx and its ecosystem with vscode plugin, cloud, generators where awesome for me, but i encountered the same problems you have, when i tried to fit it to my style it was a nightmare.

In my opinion turbo is much closer to lerna in regards of "customizability" than nx by a HUGE margin.

Once you started with nx you more or less have to use it as the configs and stuff is heavily integrated into each other and a mess. With turbo/ lerna you can use it, but dont need too.

You will be grateful in the future once you made the switch.

Btw i use pnpm as the workspace manager as i have found it to be the most fitting.
